src/minio/managers/policy_manager.py

```diff
@@ -346,6 +344,148 @@ async def delete_group_policy(self, group_name: str) -> None:
+
+    async def attach_user_policies(self, username: str) -> None:
+        """
+        Attach both home and system policies to a user.
+
+        Args:
+            username: Username to attach policies to
+
+        Raises:
+            PolicyOperationError: If policy attachment fails or policies don't exist
+        """
+        async with self.operation_context("attach_user_policies"):
+            home_policy_name = self.get_policy_name(PolicyType.USER_HOME, username)
+            system_policy_name = self.get_policy_name(PolicyType.USER_SYSTEM, username)
+
+            # Check if policies are already attached
+            home_already_attached = await self._is_policy_attached_to_target(
+                home_policy_name, PolicyTarget.USER, username
+            )
+            system_already_attached = await self._is_policy_attached_to_target(
+                system_policy_name, PolicyTarget.USER, username
+            )
+
+            # Skip if both policies are already attached
+            if home_already_attached and system_already_attached:
+                logger.info(f"Both policies already attached to user {username}")
+                return
+
+            # Validate that both policies exist before attachment
+            try:
+                home_policy = await self._load_minio_policy(home_policy_name)
+                system_policy = await self._load_minio_policy(system_policy_name)
+                if home_policy is None or system_policy is None:
+                    raise PolicyOperationError(
+                        "One or both policies do not exist or are unsupported"
+                    )
+            except Exception as e:
+                raise PolicyOperationError(
+                    f"Cannot attach policies - one or both policies do not exist: {e}"
+                ) from e
+
+            # Track attachment state for rollback
+            home_attached = home_already_attached
+            system_attached = system_already_attached
+
+            try:
+                # Attach home policy if not already attached
+                if not home_already_attached:
+                    await self.attach_policy_to_user(home_policy_name, username)
+                    home_attached = True
+                    logger.info(
+                        f"Attached home policy {home_policy_name} to user {username}"
+                    )
+
+                # Attach system policy if not already attached
+                if not system_already_attached:
+                    await self.attach_policy_to_user(system_policy_name, username)
+                    system_attached = True
+                    logger.info(
+                        f"Attached system policy {system_policy_name} to user {username}"
+                    )
+
+            except Exception as e:
+                # Implement rollback logic for partial attachment failures
+                logger.error(f"Policy attachment failed, attempting rollback: {e}")
```

Comment:
I'm not totally understanding why the rollback is important. Basically it ensures if one of the policies fails attachment, both policies are detached. But is it important that if one policy is detached, the other is also detached? I'm not really understanding the problem there.

Also, if one policy was attached prior to this method being run and attachment of the other policy fails, then the preexisting attachment will be removed, which seems odd

Reply:
👍 I updated it to only detached policies attached in this method. But I also think it's not very important. I could remove the `_rollback_newly_attached_policies` method completely.

Reply:
If it's not a big deal if there's only 1/2 policies attached (other than the obvious that the user won't be able to write to places they should be able to write to) I wouldn't bother with the complexity of rolling back

Reply:
okay I removed it.

Reply:
:+1: